### ISSUE TITLE
add finalizer usage on the new crd

### ISSUE THREAD

**Issue by clux** [kube-rs/controller-rs#24]
after #21 we should have finalizer support herein.

adapt the [`secret_syncer`](https://github.com/kube-rs/kube-rs/blob/master/examples/secret_syncer.rs) example from kube-rs to add support for finalizers to the crd herein.

if possible, it could be a compile-time feature. but that's a nice-to-have, if it ends up looking ugly, let us just have it enabled always.
setting a finalizer would require us having a non-trivial cleanup loop so we would need #21 done.
